src/main/habit-modal-dialogue/habit-modal-dialogue.component.html

```diff
@@ -0,0 +1,421 @@
+      <img [ngClass]="{'show':checkDays(daysOfWeek.Wednesday)}" alt="tick" class="daily-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Wednesday
+    </button>
+    <button (click)="toggleDay(daysOfWeek.Thursday)" mat-menu-item>
+      <img [ngClass]="{'show':checkDays(daysOfWeek.Thursday)}" alt="tick" class="daily-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Thursday
+    </button>
+    <button (click)="toggleDay(daysOfWeek.Friday)" mat-menu-item>
+      <img [ngClass]="{'show':checkDays(daysOfWeek.Friday)}" alt="tick" class="daily-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Friday
+    </button>
+    <button (click)="toggleDay(daysOfWeek.Saturday)" mat-menu-item>
+      <img [ngClass]="{'show':checkDays(daysOfWeek.Saturday)}" alt="tick" class="daily-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Saturday
+    </button>
+    <button (click)="toggleDay(daysOfWeek.Sunday)" mat-menu-item>
+      <img [ngClass]="{'show':checkDays(daysOfWeek.Sunday)}" alt="tick" class="daily-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Sunday
+    </button>
+  </mat-menu>
+  <mat-menu #interval="matMenu" [hasBackdrop]="false" class="interval-menu" overlapTrigger="false">
+    <button (click)="updateIntervalPerDays('Repeat Every 2')" mat-menu-item>
+      <img [ngClass]="{'show': checkIntervalPerDays('Repeat Every 2')}" alt="tick" class="interval-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Repeat Every 2
+    </button>
+    <button (click)="updateIntervalPerDays('Repeat Every 3')" mat-menu-item>
+      <img [ngClass]="{'show': checkIntervalPerDays('Repeat Every 3')}" alt="tick" class="interval-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Repeat Every 3
+    </button>
+    <button (click)="updateIntervalPerDays('Repeat Every 4')" mat-menu-item>
+      <img [ngClass]="{'show': checkIntervalPerDays('Repeat Every 4')}" alt="tick" class="interval-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Repeat Every 4
+    </button>
+    <button (click)="updateIntervalPerDays('Repeat Every 5')" mat-menu-item>
+      <img [ngClass]="{'show': checkIntervalPerDays('Repeat Every 5')}" alt="tick" class="interval-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Repeat Every 5
+    </button>
+    <button (click)="updateIntervalPerDays('Repeat Every 6')" mat-menu-item>
+      <img [ngClass]="{'show': checkIntervalPerDays('Repeat Every 6')}" alt="tick" class="interval-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Repeat Every 6
+    </button>
+    <button (click)="updateIntervalPerDays('Repeat Every 7')" mat-menu-item>
+      <img [ngClass]="{'show': checkIntervalPerDays('Repeat Every 7')}" alt="tick" class="interval-tick-icon"
+           src="assets/svg/tick.svg"/>
+      Repeat Every 7
+    </button>
+  </mat-menu>
+  <mat-menu #habit="matMenu" [hasBackdrop]="false" class="habit-menu" overlapTrigger="false">
```

Comment:
can we create different child components and for all the above menus and use them here then instead of adding everything in this single component

Reply:
updated